### DISCUSSION TITLE
request: add indication for 'auth required' in each lexicon api definition if auth (login) needed.

### DISCUSSION THREAD

**Discussion by itaru2622** [bluesky-social/atproto#1996] [Bluesky Lexicons]
with current lexicons definition (bluesky-social/atproto/lexicons/*) , it cannot be distinguished if each api requires authorization header or not at sending request. so please add indication if authorization (header) is required or not.

for example, 
- com.atproto.server.createSession is not required authorization header (because it create JWT).
- com.atproto.repo.createRecord is required authorization header.

but both have no indications if authorization header is required or not. it raises trouble on developing.

**Comment by bnewbold** [comment]
You are right, there isn't a clear per-endpoint way to know if auth is required. We also have a secondary form of `admin` auth, and it isn't clear when that is necessary (the `com.atproto.admin.*` namespace happens to, but there are other endpoints as well). And then there are some endpoints which don't work with app-passwords.

We'll want to have a way to indicate all of this, possibly as part of an auth refactor we are working on. But in the meanwhile we could clarify indicate in the comments/descriptions of some endpoints.

**Reply by itaru2622** [reply]
@bnewbold thank you for your comment. I posted this topic as issue https://github.com/bluesky-social/atproto/issues/2004
